### PR TITLE
Use New Attestation Receiver Method in RPC

### PR DESCRIPTION
**Pull request by rauljordan**
This is a follow-up to #3286 

---

# Description

**Write why you are making the changes in this pull request**

This PR uses the new `ReceiveAttestation` method in the RPC attester server, necessary for running with the `--next` flag.

### CODE REVIEW

File: beacon-chain/blockchain/forkchoice/service.go [hunk 1]
```diff
@@ -89,10 +89,20 @@ func (s *Store) GenesisStore(ctx context.Context, genesisState *pb.BeaconState)
...
 		return errors.Wrap(err, "could not save genesis block")
 	}
+	if err := s.db.SaveHeadBlockRoot(ctx, blkRoot); err != nil {
+		return errors.Wrap(err, "could not save head block root")
+	}
 	if err := s.db.SaveState(ctx, genesisState, blkRoot); err != nil {
 		return errors.Wrap(err, "could not save genesis state")
 	}
 
+	// For every single validator, we store their indices in the DB.
```
Comment (terencechain) [inline beacon-chain/blockchain/forkchoice/service.go]: I'm going to move this to `initializeBeaconChain` in #3290

Comment (terencechain) [reply beacon-chain/blockchain/forkchoice/service.go]: addressed in #3295

File: beacon-chain/blockchain/service.go [hunk 2]
```diff
@@ -131,23 +133,39 @@ func (c *ChainService) processChainStartTime(genesisTime time.Time, chainStartSu
...
-		return errors.Wrap(err, "could not start gensis store for fork choice")
+	if err := c.forkChoiceStore.GenesisStore(ctx, genesisState); err != nil {
+		return errors.Wrap(err, "could not start genesis store for fork choice")
 	}
 
-	c.canonicalRoots[beaconState.Slot] = c.FinalizedCheckpt().Root
+	c.headBlock = genesisBlk
+	c.headState = genesisState
+	c.canonicalRoots[genesisState.Slot] = c.FinalizedCheckpt().Root
+	c.canonicalRoots[genesisState.Slot] = c.FinalizedCheckpt().Root
```
Comment (nisdas) [inline beacon-chain/blockchain/service.go]: no need for this extra line

**Comment by terencechain** [comment]
I think this is ready but the test is failing for some reason

**Pull request merged**